src/client.rs

```diff
@@ -401,17 +395,18 @@ where
         };
 
         let targets_path = MetadataPath::from_role(&Role::Targets);
+        let targets_size = Some(targets_description.size());
```

Comment:
huh, it's unfortunate that these temporaries aren't living long enough (guessing that's why you made this change). There are merged-but-unimplemented RFCs that would fix this, but it's still disappointing.

Reply:
Yeah it was a little unfortunate, but as you see it wasn't too painful to implement, so it's not that bad.